### ISSUE TITLE
Linking libsodium with click

### ISSUE THREAD

**Issue by hameerabbasi** [kohler/click#291]
Hello, I wrote a custom element, it requires `libsodium`. Everything succeeds, except for the linking phase, where I get `undefined reference to <function>` errors. I already exported `CFLAGS` and `LDFLAGS` correctly, and made sure with `make -B -n` that the right flags were being passed... Anything else I can do?

**Comment by tbarbette** [comment]
Can we have a more extensive log?

I would look around DEPCFLAGS and CXXFLAGS also.

**Comment by pallas** [comment]
Did you try `ELEMENT_LIBS((-L/usr/local/lib -lmy_library))` as per the FAQ?

**Comment by bcronje** [comment]
In case you dont know which FAQ @pallas is referring to: [http://read.cs.ucla.edu/click/faq#one-of-my-elements-must-be-linked-against-another-library-how-can-i-make-userlevel-click-link-with-that-library]

**Comment by hameerabbasi** [comment]
Thank you, that seems to be what I'm looking for, I'll push a commit to [my repo](https://github.com/hameerabbasi/click) soon.

I did notice, however, that the command line arguments in all the makefiles were the wrong way around. [This commit](https://github.com/hameerabbasi/click/commit/462ce8427298487736319ef3154bc2d0ff5ba7b6) was what finally fixed the issue I was having. [This](https://github.com/hameerabbasi/click/blob/master/makewithadtn-ubuntu) is a simple hacked-together bash script I wrote to temporarily make it work.

In any case, I'll try to revert the makefile, try your solution and report back.

For reference I'm using vanilla Ubuntu 16.04 desktop.
